### PR TITLE
Migration/boundingbox

### PR DESCRIPTION
**Pull request by rvilarl**
BoundingBox and BoundingBoxComputation

### CODE REVIEW

File: src/boundingbox.ts [hunk 1]
```diff
@@ -3,58 +3,76 @@
 //
 // Copyright Mohit Muthanna 2010
 
+import { RenderContext } from './types/common';
+
 // Bounding boxes for interactive notation
 
 export class BoundingBox {
-  static copy(that) {
+  protected y: number;
```
Comment (0xfe) [inline src/boundingbox.ts:11]: Can you order this `x, y, w, h`?

File: src/boundingbox.ts [hunk 2]
```diff
@@ -3,58 +3,76 @@
...
+  protected x: number;
+  protected w: number;
+  protected h: number;
```
Comment (0xfe) [inline src/boundingbox.ts:14]: Why not private for all these?

File: src/boundingboxcomputation.ts [hunk 3]
```diff
@@ -10,7 +10,12 @@
 // nvg.js#L449
 
 export class BoundingBoxComputation {
-  constructor(x1, y1, x2, y2) {
+  x1: number;
+  y1: number;
+  protected x2: number;
+  protected y2: number;
```
Comment (0xfe) [inline src/boundingboxcomputation.ts:16]: Why mixed visibility here? Can't all be private?

File: src/boundingbox.ts (hunk 2 above)
Comment (rvilarl) [reply src/boundingbox.ts:14]: I thought we have decided to use `protected` in general. `private` only when `get/set` functions make a special handling like with `context`. This is here not the case and the class has also no child.

File: src/boundingboxcomputation.ts (hunk 3 above)
Comment (rvilarl) [reply src/boundingboxcomputation.ts:16]: Glyph is accessing these attributes:

https://github.com/0xfe/vexflow/blob/b373637eb2ae700a6c6cbe2606448fdea461b317/src/glyph.ts#L281

File: src/boundingbox.ts (hunk 1 above)
Comment (rvilarl) [reply src/boundingbox.ts:11]: ok

File: src/boundingbox.ts (hunk 2 above)
Comment (0xfe) [reply src/boundingbox.ts:14]: Okay, that's fine.

File: src/boundingboxcomputation.ts (hunk 3 above)
Comment (rvilarl) [reply src/boundingboxcomputation.ts:16]: I made to access functions in order to make them protected.

**Comment by rvilarl** [comment]
Ready to merge!

**Comment by 0xfe** [comment]
Looks great!

**Pull request merged**